src/Adapter/AbstractObjectModelValidator.php

```diff
@@ -78,8 +78,14 @@ protected function validateObjectModelProperty(ObjectModel $objectModel, string
      */
     protected function validateObjectModelLocalizedProperty(ObjectModel $objectModel, string $propertyName, string $exceptionClass, int $errorCode = 0)
     {
+        $localizedValues = $objectModel->{$propertyName};
+
+        if (null === $localizedValues) {
```

Comment:
I don't understand this check.. Why would the field be null, at worst it should be an empty array no?
And why ignore when it's null isn't it an error worth indicating?

Reply:
yeah so it happens because it collided with another command. 
Im not sure its safe to have this check.
Basic example: we are updating name, but description is null (because it will be updated with other command), so validator still checks every property and then tries to loop through product->description which is null.

Reply:
Then maybe it's preferable not to ignore this, because I don't think it's an expected behaviour to have fields turned into `null` by other commands 🤔
Better have an error indicating something is wrong in that case

Reply:
fields are not TURNED into null, they are simply empty, because we fill the object only partially, but validate it whole. This what happens:
1. Add product command passed tests before, because there was only manual validation implemented in handler (which checked only those fields that we fill)
2. I rebased the PR so now it uses ProductValidator, which checks whole object validity inluding multilang fields like description.
3. Now when I `AddProduct`, it only adds `name`, but validator also asserts the `description, meta_title etc.`, those fields are null as they haven't yet reached the `UpdateBasicInfo` phase to be inserted.
So Its rather implementation error than "something went wrong", because it is logical & it will happen all the time :(

Reply:
So I can see 3 paths ahead of us:

1. Current implementation: validate whole object even though not all fields have been hydrated. This means that we need, indeed, to ignore `null` fields.
2. Validate only hydrated fields by providing the exact list of what fields to validate.
3. Validate whole object after having applied modifications to it. This means a) fetch Product object b) modify fields provided by Command c) now we have a Product object with updated data and we can validate it whole. There is no `null` fields.

If I am not wrong,

Solution 1 can introduce issues because `null` is a very generic value and could be mistakenly set and this validator, however it will be catched eventually later when persisting the ObjectModel I think 🤔 ?

Solution 2 requires lot of boilerplate code to pass around list of updated fields like `['name', 'description', 'seo', ...]` => very unsatisfying solution

Solution 3 is interesting 🤔 but maybe had to implement as it means we load Product "just to validate" it ; it would actually change the whole workflow. Right now we validate first then we fetch+update the ObjectModel.

So if  `it will be catched eventually later when persisting the ObjectModel I think` is confirmed I think solution 1 seems OK

Reply:
Ok I see the problem of object partially filled, but validation as a whole

Then I would suggest relying on `ObjectModel::$update_fields` We add a getter `getFieldsToUpdate()` and we check if fieldName is inside it if not return It's a more accurate check than relying on `null` value => we check only what we modified